### DISCUSSION TITLE
Filters not appearing when using MS Search

### DISCUSSION THREAD

**Discussion by albertdiazlucas** [microsoft-search/pnp-modern-search#2526] [Q&A] [answered]
**Version used**
4.8 beta 1

**Describe the bug**
When using Microsoft Search with People entity tyoe in a search results webpart, we connect the filters webpart to the search results webpart, we configure the filters according to the data recovered by the search results webpart, but no refinement filters appear. Static filters work as expected. We cloned the 4.8 beta 1 branch and installed a debug package in order to grasp a better understanding of how the filter webpart works. We discovered that, while the MS Graph Search API request include the aggregators according to the filters we configured, the response do not include them, hence making the refinement filters unavailable.

**To Reproduce**
1. Configure a Search Results webpart with Microsoft Search with the "People" entity type
2. Perform a search to fetch some data
3. Configure a Search Filters webpart and add some refinement filters based on the search results obtained
4. Connect both webparts with the other one
5. The filters related to the search results do not appear
6. When observing the request and response from MS Graph API, aggregators are added as per configured filters, but aggregators are not returned in response

**Expected behavior**
The filters should appear when using Microsoft Search and configuring refinement filters against the properties that the search results webpart obtain

**Screenshots**

**Desktop (please complete the following information):**
 - Browser > Microsoft Edge version 107.0.1418.35

**Additional context**

**Comment by wobba** [comment]
You need to share a bit more about what properties you are trying to use as refiners/filters. The properties used has to be of type `refinable` and you need to ensure they have values. And you can verify the data/query using the Graph Explorer to test.

**Reply by albertdiazlucas** [reply]
Hi @wobba, thanks for your answer.

We have configured our filter as following:

![image](https://user-images.githubusercontent.com/9387590/200824597-5eef8c8e-bc51-403a-b28b-afc153c14115.png)

The department property is the default one that Azure AD provides, it's not a custom one. I am not currently able to check if that specific property (or the Azure AD default properties) are set as refinable or not, since I do not have access to Azure AD nor tenant administration.

About the Graph Explorer, I did try it, but it came out with the same result; request has the aggregators, but response doesn't.

Thank you very much.

**Reply by wobba** [reply]
If it doesn't work in the Graph Explorer it won't work in the web parts :) My guess is that aggregations are not supported on people via the Graph API right now. You should ask on stack overflow.

**Reply by albertdiazlucas** [reply]
Ok, thank you very much. We will check if aggregations work with other entity types and meanwhile ask on stack overflow.

**Reply by DavorSukeri** [reply]
I'm using the latest version and have the same problem. @albertdiazlucas did you resolve this?

I use Microsoft Search because I need to use companyName property as a filter...

**Comment by gretchunkim** [comment, marked as answer]
Click[ here ](https://learn.microsoft.com/en-us/graph/search-concept-aggregation) for  Microsoft reference to MS  Graph API aggregation. 


So, if the people results are in external items it is supposed to be supported, and **indeed  it's working in Graph Explore**r. 
Known limitations
Aggregations are supported only for SharePoint, OneDrive, or external items. They aren't supported for message or event types.
